tex/gregoriotex.lua

```diff
@@ -798,8 +870,8 @@ local function compile_gabc(gabc_file, gtex_file, glog_file, allow_deprecated)
     extra_args = extra_args..' -D'
   end
 
-  local cmd = string.format('%s %s -W -o %s -l %s "%s"', gregorio_exe(),
-      extra_args, gtex_file, glog_file, gabc_file)
+  local cmd = string.format('%s %s -W -o %s -l %s "%s" 2> %s', gregorio_exe(),
```

Comment:
I think this line might be causing an error for me sometimes. The error looks like this:
```
(.../tex/latex/latexconfig
/epstopdf-sys.cfg))ignored arguments: 2> tmp-gre/alma-6_0_0.glog
Usage: gregorio [OPTION]... [-s | INPUT_FILE]
Try 'gregorio --help' for more information.
Proceeding anyway...

(.../tmp-gre/alma-6_0_0.gtex
```
Sometimes it actually causes the run to pause and sometimes it finishes.

Reply:
That's supposed to be file redirection to make sure that output to stderr gets recorded in the glog file (makes debugging easier).  Not sure why it would be rendering it as "ignored arguments".  Do we need to quote the filename of the glog file?

Reply:
I can't reliably reproduce the problem (even though I've seen it many times), so I'm also stumped.

Reply:
Found the problem: if `--shell-escape` is not given, but `gregorio` is listed in `shell_escape_commands` in `texmf.cnf`, then `2>` gets interpreted as an ordinary argument. Gregorio will output a warning about extra arguments, which is harmless, but the glog file will be empty.

Reply:
`gregorio` will *always* be in the listed `shell_escape_commands` for TeX Live (and I think MiKTeX too).  I'll ask the TeX Live people about this.

Reply:
Right, which is a good thing, but presumably the restriction on redirects is a security measure. (For example, they certainly wouldn't want to allow `os.execute("gregorio | rm -rf *")`.)

Perhaps an alternative is to make gregorio itself write the .glog file.

Reply:
It already can.  That's what the `-l` option is all about.  There are just some errors that the `-l` option doesn't capture.  See #1541.

The point of the redirection is to try and capture those errors too, as they sometimes are needed to diagnose what's going wrong.

Reply:
I see, so ultimately this error message I'm seeing is unintended but mostly harmless. I've verified that if gregorio outputs an error, it does appear in the glog.

I mentioned above that sometimes TeX was stopping, and I am now guessing that it was for some other reason (still don't know what).

Reply:
So long as it's harmless, then I see no need to change things.  If it works as expected with the `--shell-escape` option, then the error messages can be captured when needed even if normal operation doesn't capture them because of the security measures.